sdk/core/core/inc/az_span.h

```diff
@@ -104,9 +140,120 @@ AZ_NODISCARD AZ_INLINE az_span az_span_from_single_item(uint8_t const * ptr) {
 }
 
 AZ_NODISCARD AZ_INLINE az_span az_str_to_span(char const * str) {
-  return (az_span){ .begin = (uint8_t const *)str, .size = strlen(str) };
+  return (az_span){
+    ._internal = { .begin = (uint8_t const *)str, .capacity = strlen(str), .length = strlen(str) }
+  };
+}
+
+/****************** Mutating az_span (used to be az_mut_span in the origins)     ******/
+
+/**
+ * @brief move the content of span @b src to @buffer and make @b out_result point to it
+ *
+ * @param buffer
+ * @param src
+ * @param out_result
+ * @return AZ_NODISCARD az_mut_span_move
+ */
+AZ_NODISCARD az_result
+az_span_move(az_span const buffer, az_span const src, az_span * const out_result);
+
+/**
+ * @brief move the content from span @b a to span @b b and viceverse.
+ * The smallest span (less content) is swapped only. Example
+ * a = 111
+ * b = 22
+ * after swaping
+ * a = 221
+ * b = 11
+ *
+ * @param a source/destination span
+ * @param b destination/source span
+ */
+void az_span_swap(az_span const a, az_span const b);
+
+/**
+ * @brief Set all the content of the span to @b fill without updating the length of the span.
+ * This is util to set memory to zero before using span or make sure span is clean before use
+ *
+ * @param span source span
+ * @param fill byte to use for filling span
+ * @return AZ_INLINE az_span_fill
+ */
+AZ_INLINE void az_span_fill(az_span const span, uint8_t const fill) {
+  if (!az_span_is_empty(span)) {
+    memset(span._internal.begin, fill, span._internal.capacity);
+  }
 }
 
+/**
+ * @brief Set one specific index of az_span from span content only
+ *
+ */
+AZ_NODISCARD AZ_INLINE az_result
+az_span_set(az_span const self, size_t const i, uint8_t const value) {
+  if (self._internal.length <= i) {
+    return AZ_ERROR_BUFFER_OVERFLOW;
+  }
+
+  self._internal.begin[i] = value;
+  return AZ_OK;
+}
+
+/**
+ * @brief converts @b src span to zero-terminated srt. Content is copied to @b buffer and then \0 is
+ * addeed at the end. Then out_result will be created out of buffer
+ *
+ * @param buffer
+ * @param src
+ * @param out_result
+ * @return AZ_NODISCARD az_mut_span_to_str
+ */
+AZ_NODISCARD az_result
+az_span_to_str(az_span const buffer, az_span const src, az_span * const out_result);
```

Comment:
I think this should now become `az_span_to_str(az_span const src, az_span * const dst)`, and as long as there's enough capacity to fit `src.length` + `'\0'`, it copies the contents and resets the length to `src.length + 1`. And as long as we end up using `memmove()` to copy contents, we're fine in terms of overlapping cases.